### PR TITLE
Fix/issue 12033 url label

### LINKED ISSUE #12033: Sidebar permalink panel: improve the URL field label
Splitting this out from #11874.

I'd tend to think the input field label `URL` is a bit confusing, as users might think they have to fill the field with a complete URL starting with `http(s)`.

screenshot 2018-11-18 at 14 27 02

I've read https://github.com/WordPress/gutenberg/issues/11858#issuecomment-438921521 and it makes perfectly sense to avoid technical terms. However, the label could be improved a bit. Also noting Squarespace does use some form of "URL" label but it also expands it adding "slug":

screenshot 2018-11-18 at 13 55 29

Other mentioned platforms use some form of "URL" as well, but at least they make it visually clear it's about the last part of the URL:

screenshot 2018-11-18 at 15 07 48

screenshot 2018-11-18 at 15 12 35

Personally, I have no strong opinions on which the best wording would be, but I'd suggest to consider a more meaningful one.

### PR DESCRIPTION
**Pull request by timwright12**
## Description
Fixes https://github.com/WordPress/gutenberg/issues/12033

## How has this been tested?
Browser
Local Unit
Local e2e

## Screenshots 
screen shot 2018-12-17 at 3 52 18 pm

## Types of changes
Bug fix/suggestion to update the permalink label to be more descriptive 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -11359,7 +11359,7 @@
 				},
 				"yargs": {
 					"version": "11.1.0",
-					"resolved": "http://registry.npmjs.org/yargs/-/yargs-11.1.0.tgz",
+					"resolved": "https://registry.npmjs.org/yargs/-/yargs-11.1.0.tgz",
```
Comment (jorgefilipecosta) [inline package-lock.json]: I think this URL changes may be unintentional? If they are intentional maybe we can do the update to https in a separate PR.

**Review by jorgefilipecosta — COMMENTED** [review]
Thank you for your contribution @timwright12. The label change seems to be working and now we see "URL Slug" as shown in your screen shot. I fear some users may not know what "slug" means but I don't have a better idea.
cc: @jasmussen, @karmatosed as they may be able to provide some ideas.

**Comment by jasmussen** [comment]
I agree "slug" isn't the best word. "Link title (URL slug)" might work better?

File: package-lock.json (hunk 1 above)
Comment (timwright12) [reply package-lock.json]: @jorgefilipecosta I think that just happened on the npm install, but I can manually revert it if it's a concern

**Comment by timwright12** [comment]
@jasmussen there are some good notes on naming in the issue from @afercia https://github.com/WordPress/gutenberg/issues/12033

This may be complicating the task, but what about a tooltip explaining what it is? To me, "link title" has the same problems as "url slug"

**Comment by jasmussen** [comment]
Yep, I'm sure there's almost a decade of history here. But that's why I suggested the full title that included both. Open to suggestions, but since we're touching this it feels like an opportunity to be slightly more verbose.

**Comment by timwright12** [comment]
@jasmussen I'm actually unfamiliar with, "Link title" as a term in that context, happy to update if others are, just want to confirm before.

@jorgefilipecosta I updated the package lock so there's no diff

**Comment by jasmussen** [comment]
Here's a little quick research. Squarespace calls it URL slug, but has a big description below:

screenshot 2018-12-21 at 08 05 23

I like the idea of a big description below regardless of what the label says.

Google Sites calls it "Custom Path":

screenshot 2018-12-21 at 08 07 52

GoDaddy's site builder doesn't have this customization as an option.

Weebly calls it "Page Permalink" but also "Pages URL":

screenshot 2018-12-21 at 08 13 17

Wix calls it Page URL:

screenshot 2018-12-21 at 08 18 42

What about calling the label "Page URL" but then elaborating in a description below?

**Comment by timwright12** [comment]
@jorgefilipecosta I'm about to head out for the holidays but I didn't want to leave this sitting, so I updated the PR to use "Page URL" and added an external link to "Learn more about permalinks" under it. What do you think about that as a solution?

**Comment by jasmussen** [comment]
Sorry about missing the added fix, and thank you for your work.

Screenshot of the current state:

screenshot 2019-01-23 at 10 18 38

Let's get a quick fix in and ship this as an improvement. But seeing this alone makes it clear that it doesn't work with just "Page URL" as the label, because we're just showing the last part. Here's what could probably work for now:

```
URL Slug:
[ test-page ]
The last part of the URL. Read more.
```

The help text below is italic, and read more links to https://codex.wordpress.org/Posts_Add_New_Screen.

Can we make that change and get this in?

**Comment by afercia** [comment]
Can we avoid "read more" links please? Links should be meaningful even when read out of context.

**Comment by jasmussen** [comment]
Sure. "Read about permalinks"?

**Comment by timwright12** [comment]
I can button this one up early next week, thanks for the feedback guys!

**Comment by timwright12** [comment]
@jasmussen @afercia This is all set!

**Comment by timwright12** [comment]
@jasmussen all set!

**Comment by jasmussen** [comment]
Woot! Put it in a milestone and we'll get it in. All it needs is a final code sanity check now.

**Comment by timwright12** [comment]
@jasmussen I don't believe I have access to add anything to a milestone, but it looks like it's assigned to one already.

**Comment by jasmussen** [comment]
Yep, I added it, so it's likely it'll land in the next plugin release.

**Comment by timwright12** [comment]
Perfect, thanks!

File: packages/edit-post/src/components/sidebar/post-link/index.js [hunk 2]
```diff
@@ -53,38 +53,46 @@ function PostLink( {
...
-						editPermalink( newValue );
-						// When we delete the field the permalink gets
-						// reverted to the original value.
-						// The forceEmptyField logic allows the user to have
-						// the field temporarily empty while typing.
-						if ( ! newValue ) {
-							if ( ! forceEmptyField ) {
+				<div className="editor-post-link">
+					<TextControl
+						label={ __( 'URL Slug:' ) }
```
Comment (talldan) [inline packages/edit-post/src/components/sidebar/post-link/index.js]: Hey @timwright12 - It looks like this accidentally changed back to 'URL Slug' in this commit https://github.com/WordPress/gutenberg/pull/12959/commits/6049de64108fc46e2d89c623bdefa9ad3c4b26ab

I think there was an agreement on 'Page URL', so I thought I'd flag it.

Comment (talldan) [reply packages/edit-post/src/components/sidebar/post-link/index.js]: Oh wait, no, I see this comment now:
https://github.com/WordPress/gutenberg/pull/12959#issuecomment-456729023

Will give it a proper test and review in that case 😄

File: packages/edit-post/src/components/sidebar/post-link/index.js [hunk 3]
```diff
@@ -53,38 +53,46 @@ function PostLink( {
...
-					value={ forceEmptyField ? '' : currentSlug }
-					onChange={ ( newValue ) => {
-						editPermalink( newValue );
-						// When we delete the field the permalink gets
-						// reverted to the original value.
-						// The forceEmptyField logic allows the user to have
-						// the field temporarily empty while typing.
-						if ( ! newValue ) {
-							if ( ! forceEmptyField ) {
+				<div className="editor-post-link">
```
Comment (talldan) [inline packages/edit-post/src/components/sidebar/post-link/index.js]: It looks like the className `editor-post-link` doesn't have any associated styles. It can probably be removed unless there's a good reason for it?

File: packages/edit-post/src/components/sidebar/post-link/index.js (hunk 2 above)
Comment (talldan) [inline packages/edit-post/src/components/sidebar/post-link/index.js]: Though my search wasn't exhaustive, it seems as though most input field labels in the sidebar don't end with a colon. The only other one I could see is 'Parent Page:'.

Might be good to drop the colon for sake of consistency. 

@jasmussen what do you think?

**Review by talldan — COMMENTED** [review]
Looks good! I had a couple of minor comments, which should be easy to resolve.

File: packages/edit-post/src/components/sidebar/post-link/index.js (hunk 2 above)
Comment (jasmussen) [reply packages/edit-post/src/components/sidebar/post-link/index.js]: That's an EXCELLENT catch & eye for detail. Yeah, if we can get that one final detail smoothed out let's ship it. Otherwise, this is also very lovely as is.

Comment (timwright12) [reply packages/edit-post/src/components/sidebar/post-link/index.js]: @talldan @jasmussen removed the colon, nice catch!

**Review by jasmussen — APPROVED** [review]
Looks good!

screenshot 2019-02-12 at 11 08 45

Thanks for this.

I'd appreciate final eyes by @talldan or others, but let's ship this!

File: packages/edit-post/src/components/sidebar/post-link/index.js (hunk 3 above)
Comment (youknowriad) [reply packages/edit-post/src/components/sidebar/post-link/index.js]: Is this div really necessary or should we just replace it with `<Fragment />`
